CONTRIBUTING.md

```diff
@@ -186,6 +186,63 @@ guidelines:
 - If you're not sure what to name it, pick any name and we can help you
   with it.
 
+### Borrowing Request Types
+
+In order to reduce memory allocations required for clients to send requests
+to a server, we support borrowed types for request types. (We hope to support
+borrowing in response types sometime, but that is dependent on GATs becoming
+stable.)
+
+#### Types to borrow
+
+| Field type  | Borrowed type        | Notes |
+| ----------  | -------------        | ----- |
+| strings     | `&'a str`            |       |
+| identifiers | `&'a IdentifierType` |       |
+| `Vec<_>     ` | `&'a [_]`            | The inner type should not be borrowed. |
```

Comment:
```suggestion
| `Vec<_>`    | `&'a [_]`            | The inner type should not be borrowed. |
```

Reply:
I'm also not super sure about this one. But I guess it's good to have docs for how things currently work, even if we change policies later on.

Reply:
Yeah, agreed. Do you want to open up a discussion issue to track that, or just let it sit for now?

Reply:
I'll open an issue.